bika/lims/browser/resultsimport/resultsimport.py

```diff
@@ -0,0 +1,252 @@
+                                         'Parser not found...', file_name)
+                        continue
+                    # We will run import with some default parameters
+                    # Expected to be modified in the future.
+                    logger.info('Parsing ' + file_name)
+                    parser = parser_function(result_file)
+                    importer = GeneralImporter(
+                                parser=parser,
+                                context=self.portal,
+                                idsearchcriteria=['getRequestID',
+                                                  'getSampleID',
+                                                  'getClientSampleID'],
+                                allowed_ar_states=['sample_received'],
+                                allowed_analysis_states=None,
+                                override=[False, False],
+                                instrument_uid=i.UID())
+                    tbex = ''
+                    try:
+                        importer.process()
+                    except:
+                        tbex = traceback.format_exc()
+                    errors = importer.errors
+                    logs = importer.logs
+                    if tbex:
+                        errors.append(tbex)
+                    final_log = ''
+                    success_log = self.getInfoFromLog(logs, 'Import finished')
+                    if success_log:
+                        final_log = success_log
+                    else:
+                        final_log = errors
+                    self.insert_file_name(folder, file_name)
+                    self.add_to_logs(i, interface, final_log, file_name)
+                    self.add_to_log_file(i.Title(), interface, final_log,
+                                         file_name, folder)
+        logger.info('End of auto import...')
+        return 'Auto-Import finished...'
+
+    def getAlreadyImportedFiles(self, folder):
+        try:
+            with open(folder+'/imported.csv', 'r') as f:
+                imported = f.readlines()
+                imported = [i.strip() for i in imported]
+                return imported
+        except:
+            with open(folder+'/imported.csv', 'w') as f:
+                f.write('imported.csv\n')
+                f.write('logs.log\n')
+                f.close()
+            return ['']
+        return None
+
+    def insert_file_name(self, folder, name):
+        try:
+            with open(folder+'/imported.csv', 'a') as fd:
+                fd.write(name+'\n')
+                fd.close()
+        except:
+            pass
+
+    def getInfoFromLog(self, logs, keyword):
+        try:
+            for log in logs:
+                if keyword in log:
+                    return log
+            return None
+        except:
+            return None
+
+    def add_to_logs(self, instrument, interface, log, filename):
+        if log:
```

Comment:
There is no need to create an AutoImportLog if the log is empty I guess, so instead of

    if log:
        log = ''.join(log)
        log = log[:80]+'...' if len(log) > 80 else log

do:

    if not log:
        return
    log = ''.join(log)
    log = log[:80]+'...' if len(log) > 80 else log